docs/blog/2022-10-04-python2to3.md

```diff
@@ -0,0 +1,55 @@
+
+1. [*Set up a custom Python 2 environment*](2022-05-11-python2to3.md#1-set-up-a-custom-python-2-environment-using-conda-or-containers)
+2. [*Find an alternative tool that fits your needs*](2022-05-11-python2to3.md#2-find-an-alternative-tool-that-fits-your-needs)
+3. [*Convert Python 2 code to Python 3*](2022-05-11-python2to3.md#3-convert-python-2-code-to-python-3)
+
+
+## 1. Set up a custom Python 2 environment using Conda or Containers
+
+It is best to create this python environment within [conda](https://docs.conda.io/en/latest/). 
+This conda environment can be made even more portable by using [Docker](https://www.docker.com) or [Apptainer](https://apptainer.org).
+Currently, [Eagle only supports Apptainer](https://nrel.github.io/HPC/Documentation/Software_Tools/Containers/).
+Subsequently, the outputs of the Python 2 code can be incorporated, e.g., using a file-based approach.
+
+It is possible to run legacy Python 2 code in parallel within a container or a conda environment, but your mileage may vary.
+
+
+## 2. Find an alternative tool that fits your needs
+
+There are usually multiple alternatives to your software of concern that have the same or slightly different features.
+Some of the potential advantages and disadvantages include
+
+**Advantages**
+
+   1. If they are written in Python 3, they are newer and might offer better software support.
+   2. They allow for efficient parallelism, enabling better use of Eagle.
+   3. Leverages Python 3 features and packages, e.g., Abstract Base Classes.
+
+**Disadvantages**
+
+   1. It may break your build environment.
+   2. It may require a significant code rewrite to extract the best performance.
+   3. Additional code may need to be written if the alternative does not have all the features as the Python 2 software.
+
+
+## 3. Convert Python 2 code to Python 3
+
+This is an option when the Python 2 code under consideration does not have a lot of dependencies, you have the source code, and the software license allows you to make changes to the source code. 
+For scientific software, we do encourage developers to make this version jump as this enables code longevitiy and accessibility.
+Several online resource are available to enable porting your code to Python 3.
+Some of them include:
+
+   1. [Porting Python 2 Code to Python 3](https://docs.python.org/3/howto/pyporting.html)
+   2. [THe Conservative Python 3 Porting Guide](https://portingguide.readthedocs.io/en/latest/)
+   3. [Supporting Python 3: An in-depth guide](http://python3porting.com)
+   4. [Python FAQ: How do I port to Python 3?](https://eev.ee/blog/2016/07/31/python-faq-how-do-i-port-to-python-3/)
+   5. [How to Port Python 2 Code to Python 3](https://www.digitalocean.com/community/tutorials/how-to-port-python-2-code-to-python-3)
+
+Note that the above is not an exhasutive list and we encourage teh user to look at other resources as well.
```

Comment:
Typo: 'teh'

Reply:
Fixed typo in the most recent push and added entry to `mkdocs.yml`.